### PR TITLE
Issue 21272 contetlet as json views

### LINKED ISSUE #21272: Contentlet as Json include system fields
In Order to centralize the JSON generation, We have decided that the new immutable contetlet that is getting stored into the DB from now on must include all system fields. 

The system fields must not be stored as their raw representation they have to have more info so they look as they are hydrated when retrieved by our APIs

- For example, an image field should include the resource link. 
- Binary fields must include additional info from the Metadata at least SHA-256 and Mime-Type. So they can be located regardless of their storage medium.
- Categories must be stored as an array.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This brings additional hydration features to the new immutable contetlet
a few changes were made to the internal check-in method. adding an extra save operation that saves again the JSON once all system fields are ready. 

Some of the new properties hydrated in the json are related to the metadata. Which originally was written down the indexing process which happens before the check-in. In order to include this info in the JSON, the metadata has to be generated earlier.  And it needs to be generated after the binaries are sorted out.

Basically, check-in saves one initial time to create the entry another time to set the identifier and now this third time is required to update the contetlet as JSON once all the system fields are generated. 
An Example of why this situation is that 
Binaries are generated until after the contetlet has been successfully saved before that moment the binaries are just temp files.

### CODE REVIEW

**Comment by fabrizzio-dotCMS** [comment]
closed in favor of https://github.com/dotCMS/core/pull/21585
